spec/connectors/registry_spec.rb

```diff
@@ -9,21 +9,79 @@
 require 'connectors/registry'
 
 describe Connectors::Factory do
-  let(:factory) do
-    described_class.new
+  subject { described_class.new }
+
+  let(:configuration) {
+    {}
+  }
+
+  let(:job_description) {
+    {}
+  }
+
+  let(:registered_connector) {
+    'my-connector'
+  }
+
+  let(:unregistered_connector) {
+    'another-connector'
+  }
+
+  class MyConnector
+    def initialize(configuration: {}, job_description: {})
+      # nothing needed
+    end
+  end
+
+  before(:each) do
+    subject.register(registered_connector, MyConnector)
   end
 
   describe '#connector_class' do
-    it 'let us register a connector under a name' do
-      class MyConnector
-        def works
-          'works'
-        end
+    it 'returns registered class' do
+      expect(subject.connector_class(registered_connector)).to eq MyConnector
+    end
+  end
+
+  describe '#registered?' do
+    context 'my-connector is registered' do
```

Comment:
I would probably rename this style of sections throughout the file to something like:

```
context 'when called against previously registered service type'
```

It's also a good practice to start statements inside `context` with "when": "when connector is registered", "when variable is nil", etc.